### ISSUE TITLE
Support for Datastore Cluster with vSphere provider?

### ISSUE THREAD

**Issue by haukurk** [kubernetes/kubernetes-anywhere#387]
Has anyone used SDRS (Datastore cluster) with the vSphere provider?

I'm getting the following error when provisioning:

```
* vsphere_virtual_machine.kubevm3: ServerFaultCode: A specified parameter was not correct: Datastore unspecified for at least one disk in SDRS-disabled VM
```

**Comment by haukurk** [comment]
Managed to get it working by using a LUN inside the DRS cluster using "DatastoreCluster/LUN" as the datastore parameter when provisioning.

**Comment by divyenpatel** [comment]
@haukurk what is the version of Kubernetes you are using?
We recently made some changes for Datastore Cluster support.

See PRs
https://github.com/kubernetes/kubernetes/pull/44868
https://github.com/kubernetes/kubernetes/pull/45240

ping us if you come across any issue.

@BaluDontu @tusharnt

**Comment by BaluDontu** [comment]
@haukurk : Also, we have recently added the support for dynamic volume provisioning for datastores with in datastore cluster. You can see more about it in - https://github.com/kubernetes/kubernetes/tree/master/examples/volumes/vsphere

This will be available in next kubernetes release 1.6.x.
